source/shared/core_stmt.cpp

```diff
@@ -1171,15 +1165,22 @@ void core_sqlsrv_set_query_timeout( sqlsrv_stmt* stmt, long timeout TSRMLS_DC )
         
         // a query timeout of 0 indicates "no timeout", which means that lock_timeout should also be set to "no timeout" which 
         // is represented by -1.
-        long lock_timeout = (( timeout == 0 ) ? -1 : timeout * 1000 /*convert to milliseconds*/ );
+        int lock_timeout = (( timeout == 0 ) ? -1 : timeout * 1000 /*convert to milliseconds*/ );
 
         // set the LOCK_TIMEOUT on the server.
         char lock_timeout_sql[ 32 ];
+#ifndef _WIN32		
+        int written = snprintf( lock_timeout_sql, sizeof( lock_timeout_sql ), "SET LOCK_TIMEOUT %d", 
```

Comment:
We're using format characters in `SET LOCK_TIMEOUT %d`, however we can get rid of `ifdef` by using snprintf for both platforms.